stripe/_api_requestor.py

```diff
@@ -316,15 +323,60 @@ def handle_error_response(self, rbody, rcode, resp, rheaders) -> NoReturn:
             )
 
         if err is None:
-            err = self.specific_api_error(
-                rbody, rcode, resp, rheaders, error_data
+            err = (
+                self.specific_v2_api_error(
+                    rbody, rcode, resp, rheaders, error_data
+                )
+                if api_mode == "V2"
+                else self.specific_v1_api_error(
+                    rbody, rcode, resp, rheaders, error_data
+                )
             )
 
         raise err
 
-    def specific_api_error(self, rbody, rcode, resp, rheaders, error_data):
+    def specific_v2_api_error(self, rbody, rcode, resp, rheaders, error_data):
+        type = error_data.get("type")
+        code = error_data.get("code")
+        message = error_data.get("message")
+        error_args = {
+            "message": message,
+            "http_body": rbody,
+            "http_status": rcode,
+            "json_body": resp,
+            "headers": rheaders,
+            "code": code,
+        }
+
         log_info(
-            "Stripe API error received",
+            "Stripe v2 API error received",
+            error_code=code,
+            error_type=error_data.get("type"),
+            error_message=message,
+            error_param=error_data.get("param"),
+        )
+
+        if type == "idempotency_error":
+            return error.IdempotencyError(
+                message,
+                rbody,
+                rcode,
+                resp,
+                rheaders,
+                code,
+            )
+        # switchCases: The beginning of the section generated from our OpenAPI spec
+        elif type == "temporary_session_expired":
+            return error.TemporarySessionExpiredError(**error_args)
+        # switchCases: The end of the section generated from our OpenAPI spec
+
+        return self.specific_v1_api_error(
```

Comment:
Shouldnt we look for code == 'invalid_fields' here before falling back on v1 errors?